lib/taskana-core/src/main/java/pro/taskana/user/api/UserService.java

```diff
@@ -0,0 +1,88 @@
+package pro.taskana.user.api;
+
+import pro.taskana.common.api.exceptions.InvalidArgumentException;
+import pro.taskana.common.api.exceptions.NotAuthorizedException;
+import pro.taskana.user.api.exceptions.UserAlreadyExistException;
+import pro.taskana.user.api.exceptions.UserNotFoundException;
+import pro.taskana.user.api.models.User;
+
+/** The UserServices manages all operations concerning {@linkplain User Users}. */
+public interface UserService {
+
+  /**
+   * Returns a new {@linkplain User} with all fields being empty.
+   *
+   * <p>It will be only generated but not yet inserted until method {@linkplain
+   * UserService#createUser(User)} is called.
+   *
+   * @return the new {@linkplain User}
+   */
+  User newUser();
+
+  /**
+   * Gets a {@linkplain User}.
+   *
+   * <p>If a {@linkplain User} with the specified {@linkplain User#getId() id} is existing in the
+   * database, it is returned.
+   *
+   * @param id the {@linkplain User#getId() id} of the {@linkplain User} to be retrieved
+   * @return the retrieved {@linkplain User}
+   * @throws UserNotFoundException if there does not exist a {@linkplain User} with the specified
+   *     {@linkplain User#getId() id} inside the database
+   */
+  User getUser(String id) throws UserNotFoundException;
+
+  /**
+   * Creates a new {@linkplain User}.
+   *
+   * <p>If all checks have passed the specified {@linkplain User} will be inserted in the database.
+   * The {@linkplain User#getId() id} of the {@linkplain User} must be set and should be unique,
+   * such that no other {@linkplain User} with same {@linkplain User#getId() id} is already existing
+   * in the database. The {@linkplain User#getFirstName() first name} and the {@linkplain
+   * User#getLastName() last name} must not be null. The fields for the {@linkplain
+   * User#getFullName() full name} and the {@linkplain User#getLongName() long name} are set
+   * according to following rules:
+   *
+   * <ul>
+   *   <li><b>fullName</b> = lastName, firstName
+   *   <li><b>longName</b> = lastName, firstName - (id)
+   * </ul>
+   *
+   * @param userToCreate the {@linkplain User} which should be inserted
+   * @return the inserted {@linkplain User}
+   * @throws InvalidArgumentException if some fields are not set properly
+   * @throws NotAuthorizedException if the current user is no admin or business-admin
```

Comment:
I made here the difference of current user to {@linkplain User} because one is the person in charge of the current operations and one is an entity existing in TASKANA.